### PR TITLE
Use new data format

### PR DESCRIPTION
**Pull request by mrchtr**
This PR applies the usage of the new data format:

- fixes all tests
- update component specifications and component code
- remove subset field usage in `pipeline.py`

### CODE REVIEW

File: components/chunk_text/tests/chunk_text_test.py [hunk 1]
```diff
@@ -3,11 +3,11 @@
 from src.main import ChunkTextComponent
 
 
-def test_transform():
+def test_transform__():
```
Comment (RobbeSneyders) [inline components/chunk_text/tests/chunk_text_test.py:6]: Why the `__` suffix?

File: components/embed_text/fondant_component.yaml [hunk 2]
```diff
@@ -40,12 +36,12 @@ args:
       Pass only the keys required by the model provider or conveniently pass all keys you will ever need. 
       Pay attention how to name the dictionary keys so that they can be used by the model provider.
     type: dict
-    default: {}
+    default: { }
```
Comment (RobbeSneyders) [inline components/embed_text/fondant_component.yaml:39]: Can you turn this off in your formatter?

File: components/embedding_based_laion_retrieval/fondant_component.yaml [hunk 3]
```diff
@@ -7,18 +7,14 @@ tags:
...
+      type: float32
 
 produces:
-  images:
-    fields:
-      url:
-        type: string
+  images_url:
+    type: string
   additionalSubsets: false
```
Comment (RobbeSneyders) [inline components/embedding_based_laion_retrieval/fondant_component.yaml]: You might want to comment this one out as well.

File: components/image_cropping/src/main.py [hunk 4]
```diff
@@ -46,12 +46,12 @@ def __init__(
...
     def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
         # crop images
-        dataframe["images"]["data"] = dataframe["images"]["data"].apply(
+        dataframe["images_data"] = dataframe["images_data"].apply(
             lambda image: remove_borders(image, self.cropping_threshold, self.padding),
         )
 
         # extract width and height
-        dataframe["images"][["width", "height"]] = dataframe["images"]["data"].apply(
+        dataframe["images_width", "images_width"] = dataframe["images_data"].apply(
```
Comment (RobbeSneyders) [inline components/image_cropping/src/main.py:54]:
```suggestion
        dataframe["images_width", "images_height"] = dataframe["images_data"].apply(
```

File: src/fondant/core/manifest.py [hunk 5]
```diff
@@ -238,7 +238,7 @@ def remove_field(self, name: str) -> None:
 
         del self._specification["fields"][name]
 
-    def evolve(  # noqa : PLR0912 (too many branches)
+    def evolve(  # : PLR0912 (too many branches)
```
Comment (RobbeSneyders) [inline src/fondant/core/manifest.py:241]: Something happened to the comment here.

File: src/fondant/pipeline/pipeline.py [hunk 6]
```diff
@@ -443,50 +443,36 @@ def _validate_pipeline_definition(self, run_id: str):
...
-                                f"'{component_spec.name}' component does not match  the "
-                                f"previously created subset definition.\n The '{field_name}' "
-                                f"field is currently defined with the following schema:\n"
-                                f"{manifest_fields[field_name]}\nThe current component to "
-                                f"trying to invoke it with this schema:\n{subset_field}"
-                            )
-                            raise InvalidPipelineDefinition(
-                                msg,
-                            )
+                        # Get the corresponding manifest fields
```
Comment (RobbeSneyders) [inline src/fondant/pipeline/pipeline.py:459]:
```suggestion
                    # Get the corresponding manifest fields
```

**Review by RobbeSneyders — COMMENTED** [review]
Thanks @mrchtr!

All the test files in `tests/pipeline/examples are shown as newly added. I think this is because you didn't remove the old ones (or at least didn't include this in git).

I would be in favor of choosing better column names. Eg.
- `images_data` -> `image`
- `captions_data` -> `caption`

But for me it's fine to include it like this for now. We can change them later.

File: components/chunk_text/tests/chunk_text_test.py (hunk 1 above)
Comment (mrchtr) [reply components/chunk_text/tests/chunk_text_test.py]: PyCharm related bug. For some reasons I can't execute different tests with the same name in different folders.

**Pull request merged**